### ISSUE TITLE
Various problems with category pages

### ISSUE THREAD

**Issue by TheNetStriker** [openzim/mwoffliner#829]
I've just tested the new --getCategories function and found a few issues:

1) Links to pages that contain slashes and points do not work. (e.g. "Page/Subpage" or "This.That") They do not seem to contain a link at all.
2) Links to pages that contain slashes do also not work on regular pages. I guess the problem is that the /A/ part is missing in the link. (e.g. zim://Page/Subpage instead of zim://A/Page/Subpage)
3) Page links on the category page are not in alphabetical order.
4) The sort keyword of the categories is not taken into account. (e.g. a page with the name Some:Page with the category tag [[Category:Test|Page]] is listed under the letter S instead of P in the Test category)

If those problems could be fixed the category funtion would be perfect for me.

**Comment by kelson42** [comment]
@TheNetStriker Thank you for your feedback about categories, this is one of the first one and like you see this still a work-in-progress. I'll have a look in detail to what you have reported and might split-up the whole in many smaller tickets.

**Comment by rgaudin** [comment]
Here a some additions based on [this bm zim](http://download.kiwix.org/zim/wikipedia/wikipedia_bm_all_maxi_2019-09.zim)

Category NS Alias: `Catégorie`

* Links to categories from bottom of articles are OK.
* `/U/Catégorie:Kan`
  * https://bm.wikipedia.org/wiki/Cat%C3%A9gorie:Kan
  * Under **J** should be only one link to sub-category **Jamu** (`/U/Catégorie:Jamu`)
  * Instead we have five links to that sub category.
  * Links to Pages are OK.
* `/U/Catégorie:Dunia`
  * https://bm.wikipedia.org/wiki/Cat%C3%A9gorie:Dunia
  * Links to sub-categories are OK (single sub appears twice instead of once though)
  * Pages lists the 2 pages present online plus a link to the sub-category.
  * That link is wrong (`/A/` instead of `/U/`)
* `/U/Catégorie:Afrika`
  * https://bm.wikipedia.org/wiki/Cat%C3%A9gorie:Afrika
  * Link in the text intro (not sub-categories section) goes to `/U/` instead of `/A/`
  * Links to sub-categories with no page (only sub categories) are absent

I also tested on [this ar zim](http://download.kiwix.org/zim/other/mawsouaa_ar_all_2019-09.zim) corresponding to [mawsouaa.tn](http://www.mawsouaa.tn)

Unfortunately this wiki is offline ATM so we can't compare. Leaving those comments still as there might by an RTL-related issue.

Category NS Alias: `تصنيف`

* Links to categories in homepage not detected as such, pointing to `/A/`.
* No bottom of article categories in any article.
* All category pages are empty (contains only copyright footer). ex: `/U/تصنيف%3Aالعلوم`

**Comment by ImportTaste** [comment]
> * Links to categories in homepage not detected as such, pointing to `/A/`.
> * No bottom of article categories in any article.
> * All category pages are empty (contains only copyright footer). ex: `/U/تصنيف%3Aالعلوم`

It's not a RTL issue, I'm getting the same problem with https://aids.miraheze.org/

The articles are on /A/ but the categories are on /U/, and there's no page content other than the footer.

At least it actually did save most of the articles though, I couldn't get the tool to work with https://naidb.miraheze.org/ at all. Lots of miraheze.org issues, it seems.

**Comment by benoit74** [comment]
All these issues are either not relevant anymore (we do not use old namespaces anymore) and/or fixed with #2617 

Nothing left to fix in this issue. Please open another dedicated issue if you still find a problem (or reopen this issue if it is strongly related)

Note that #2617 is ought to be released in 2.0.0 (hence not yet released) and pagination (category with more than 200 pages) is not yet implemented (#2620).

**Comment by God-damnit-all** [comment]
> (we do not use old namespaces anymore)

If this is in reference to older versions of MediaWiki, do bear in mind that the ones that tend to need scraping the most are the ones that haven't been maintained in quite some time and need their data rescued.

**Comment by benoit74** [comment]
Nope, this is a reference to ZIM [namespaces](https://wiki.openzim.org/wiki/ZIM_file_format#Namespaces) which have changed quite a long time ago and allowed to not have `/A/` and `/U/` in kiwix-serve URLs for instance
